web/src/client/network/index.js

```diff
@@ -156,48 +187,204 @@ o  *   NetworkManagerAdapter.
    * @param {object} options - connection options
    */
   async addAndConnectTo(ssid, options) {
-    return this.adapter.addAndConnectTo(ssid, options);
+    // duplicated code (see network manager adapter)
+    const wireless = { ssid };
+    if (options.security) wireless.security = options.security;
+    if (options.password) wireless.password = options.password;
+    if (options.hidden) wireless.hidden = options.hidden;
+
+    const connection = createConnection({
+      id: ssid,
+      wireless
+    });
+
+    // the connection is automatically activated when written
+    return this.addConnection(connection);
   }
 
   /**
    * Adds a new connection
    *
+   * If a connection with the given ID already exists, it updates such a
+   * connection.
+   *
    * @param {Connection} connection - Connection to add
+   * @return {Promise<Connection>} the added connection
    */
   async addConnection(connection) {
```

Comment:
NP: What do think about renaming this method to `AddOrUpdateConnection` ?

Reply:
I thought about that, but it is weird having an `addOrUpdateConnection` and `updateConnection.` Perhaps we might need to merge them and have something like `saveConnection` only. I will think about that.